algorithms/kernel/decision_tree/decision_tree_classification_split_criterion.i

```diff
@@ -448,31 +448,31 @@ struct GiniWeighted
         const ValueType invLeftWeight = static_cast<ValueType>(1) / leftWeight;
         const ValueType leftProbability = leftWeight * invTotalWeight;
         const ValueType rightProbability = rightWeight * invTotalWeight;
-        ValueType leftGini = 1;
-        ValueType rightGini = 1;
+        ValueType leftGini = leftWeight * leftWeight;
```

Comment:
it`s not obvious why initial value was changed.
Could you add comment please?

Reply:
@ShvetsKS 
gini index can be rewritten in this way:
![image](https://user-images.githubusercontent.com/50139720/69537330-f64a6780-0f90-11ea-9fab-8142b584872c.png)
where W - sum of weights of all classes, w_k - weight of k-th class, C - number of classes